boards/beagle/pocketbeagle_2/doc/index.rst

```diff
@@ -0,0 +1,107 @@
+.. zephyr:board:: pocketbeagle_2
+
+Overview
+********
+
+PocketBeagle 2 is a computational platform powered by TI AM62x SoC (there are two
+revisions, AM6232 and AM6254).
+
+The board configuration provides support for the ARM Cortex-M4F MCU core and the
+following features:
+
+- Nested Vector Interrupt Controller (NVIC)
+- System Tick System Clock (SYSTICK)
+
+The board configuration also enables support for the semihosting debugging console.
+
+See the `PocketBeagle 2 Product Page`_ for details.
+
+Hardware
+********
+PocketBeagle 2 features the TI AM62x SoC based around an Arm Cortex-A53 multicore
+cluster with an Arm Cortex-M4F microcontroller, Imagination Technologies AXE-1-16
```

Comment:
If the board revisions are already announced with AM6232 and AM6254 on different board revisions (as mentioned in comments by @jadonk and @RobertCNelson ), I will recommend it to be added and documented as per https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html